### ISSUE TITLE
"failed to verify vote with ChainID" happens occasionally

### ISSUE THREAD

**Issue by yihuang** [tharsis/ethermint#140]
__System info:__ [ethermint(6be18a1cb487b1a84b78c52c8e6e28ad318c799a), macos]

__Steps to reproduce:__

1. setup local devnet
2. repeatedly stop and start the node process
3. occasionally, there's this error message, and chain halted.

__Expected behavior:__ chain should continue generating blocks without error

__Actual behavior:__ chain halted with error message

__Additional info:__

The surrounding logs:
 ```
chainmaind-1/node1.log I[2021-06-17|14:25:50.217] Replay: Done                                 module=consensus
chainmaind-1/node0.log I[2021-06-17|14:25:50.282] Replay: Timeout                              module=consensus height=70 round=0 step=RoundStepPropose dur=3s
chainmaind-1/node1.log I[2021-06-17|14:25:50.217] Starting TimeoutTicker service               module=consensus impl=TimeoutTicker
chainmaind-1/node0.log I[2021-06-17|14:25:50.282] Replay: New Step                             module=consensus height=70 round=0 step=RoundStepPropose
chainmaind-1/node0.log I[2021-06-17|14:25:50.282] Replay: Done                                 module=consensus
chainmaind-1/node0.log I[2021-06-17|14:25:50.282] Starting TimeoutTicker service               module=consensus impl=TimeoutTicker
chainmaind-1/node0.log I[2021-06-17|14:25:50.304] failed attempting to add vote                module=consensus err="failed to verify vote with ChainID chainmaind-1 and PubKey PubKeyEd25519{5B1F85332A2752A8B27121996A4EAA1B5F17CEBA62979E55B2518FC43BE3A33E}: invalid signature"
chainmaind-1/node0.log E[2021-06-17|14:25:50.304] failed to process message                    module=consensus height=70 round=0 peer= err="error adding vote" msg="[Vote Vote{1:D148C82AFD19 70/00/SIGNED_MSG_TYPE_PREVOTE(Prevote) B05552E3E8A3 7B8806DED001 @ 2021-06-17T06:25:50.281127Z}]"
chainmaind-1/node1.log I[2021-06-17|14:25:53.217] Timed out                                    module=consensus dur=3s height=70 round=0 step=RoundStepPropose
chainmaind-1/node0.log I[2021-06-17|14:25:53.284] Timed out                                    module=consensus dur=3s height=70 round=0 step=RoundStepPropose
```

**Comment by JayT106** [comment]
@yihuang would like to confirm the reproduction steps
after init the testnet env (with 2 nodes), I ran
1. pystarport start --config config.yaml
2. ctrl + c to stop
3. repeat steps 1&2 until seeing the error

**Comment by JayT106** [comment]
no luck to reproduce it, but from the code tracing, looks like it might happen under the `bad peer connection` or `Unexpected step` status.
https://github.com/tendermint/tendermint/pull/3871/files#diff-51109119207cf15f4491c8288086704e22a037357ad6ebc68a3b24b4818904fdR1521

**Comment by yihuang** [comment]
> @yihuang would like to confirm the reproduction steps
> after init the testnet env (with 2 nodes), I ran
> 
> 1. pystarport start --config config.yaml
> 2. ctrl + c to stop
> 3. repeat steps 1&2 until seeing the error

yes, similar like that, in development, I repeatedly do edit rebuild restart cycle, I've seen this issue for several times. Not sure modify ethermint code contributes to this issue.

**Comment by JayT106** [comment]
@fedekunze could you assign this issue to me? I am investigating it. Thanks.

**Comment by yihuang** [comment]
ref: https://github.com/tendermint/tendermint/issues/6738

**Comment by JayT106** [comment]
the fix has been bp to 0.34.x
https://github.com/tendermint/tendermint/pull/6783

**Comment by fedekunze** [comment]
closing
